### PR TITLE
feat: add docs for installing locally using kind

### PR DESCRIPTION
**Pull request by coadler**

### CODE REVIEW

File: setup/local-preview.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+In a Mac or Linux environment, ensure that the following tools are installed:
+
+1. `kubectl`: [https://kubernetes.io/docs/tasks/tools/install-kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl)
+1. `helm`: [https://helm.sh/docs/intro/install](https://helm.sh/docs/intro/install)
+1. `kind`: [https://kind.sigs.k8s.io/docs/user/quick-start/#installation](https://kind.sigs.k8s.io/docs/user/quick-start/#installation)
+
+Then, use the following script to install Coder:
+
+```bash
+curl -fsSL https://try.coder.com | PORT="8080" sh
```
Comment (nathanpotter) [inline setup/local-preview.md:27]: I'm assuming yes for this, but has this script been verified with mac's default old bash version? I ran it remotely earlier, but I didn't check on my MacBook. I can check tomorrow if it hasn't been tested

**Review by nathanpotter — APPROVED** [review]
LGTM🎉

File: setup/local-preview.md (hunk 1 above)
Comment (coadler) [reply setup/local-preview.md:27]: I don't believe I'm using anything special so it's unlikely that it wouldn't work, but definitely wouldn't hurt to give it a try.

Comment (nathanpotter) [reply setup/local-preview.md:27]: It took a while to spin up the ingress, and then it failed with:
```
error: timed out waiting for the condition on pods/timescale-0
```

Comment (nathanpotter) [reply setup/local-preview.md:27]: timed out when retrying as well :(

**Comment by bpmct** [comment]
@coadler I'm not sure if this is related to my setup or the install script, but I noticed an error running CVMs on my deployment. If I have to update Docker/Kind config to support the new kernel version/seccomp API, can we document that here?

# http://localhost:8080/environments:
```

{"ts":"2021-01-31T13:41:18.0194895Z","level":"INFO","msg":"log","logger_name":"sysbox-fs","fields":{"content":"time=\"2021-01-31 13:41:18\" level=fatal msg=\"syscallMonitorService initialization error (Error: unsupported kernel). Exiting ...\""}}
{"id":"","environment_id":"","build_id":"","time":"0001-01-01T00:00:00Z","type":"yield-fail","msg":"Sysbox exited, possibly because of an unsupported kernel version. Please contact an infrastructure administrator and request a node kernel with seccomp API level \u003e= 5."}
{"ts":"2021-01-31T13:41:18.0209707Z","level":"FATAL","msg":"sysbox-fs exited, possibly because of an unsupported kernel version. Please contact an infrastructure administrator and request a node kernel with seccomp API level \u003e= 5.","fields":{"error":"exit status 1"}}
```


```bash
➜  scripts git:(master) ✗ docker --version
Docker version 20.10.2, build 2291f61
➜  scripts git:(master) ✗ kind --version
kind version 0.8.1
➜  scripts git:(master) ✗ 
```

**Comment by coadler** [comment]
> @coadler I'm not sure if this is related to my setup or the install script, but I noticed an error running CVMs on my deployment. If I have to update Docker/Kind config to support the new kernel version/seccomp API, can we document that here?
> 
> http://localhost:8080/environments:
> 
> ```
> 
> {"ts":"2021-01-31T13:41:18.0194895Z","level":"INFO","msg":"log","logger_name":"sysbox-fs","fields":{"content":"time=\"2021-01-31 13:41:18\" level=fatal msg=\"syscallMonitorService initialization error (Error: unsupported kernel). Exiting ...\""}}
> {"id":"","environment_id":"","build_id":"","time":"0001-01-01T00:00:00Z","type":"yield-fail","msg":"Sysbox exited, possibly because of an unsupported kernel version. Please contact an infrastructure administrator and request a node kernel with seccomp API level \u003e= 5."}
> {"ts":"2021-01-31T13:41:18.0209707Z","level":"FATAL","msg":"sysbox-fs exited, possibly because of an unsupported kernel version. Please contact an infrastructure administrator and request a node kernel with seccomp API level \u003e= 5.","fields":{"error":"exit status 1"}}
> ```
> 
> ```shell
> ➜  scripts git:(master) ✗ docker --version
> Docker version 20.10.2, build 2291f61
> ➜  scripts git:(master) ✗ kind --version
> kind version 0.8.1
> ➜  scripts git:(master) ✗ 
> ```

@bpmct Unfortunately Docker Desktop downgraded to kernel v4.9, so you'll have to use https://docs.docker.com/docker-for-mac/release-notes/#docker-desktop-community-2501 which is the last release that uses kernel v5. I'll put that in the docs.

**Comment by bpmct** [comment]
Would it be possible to also mention to uses that DevURLs won't work by default, but they could add a tool like [ngrok](https://ngrok.com/) to their images to test their applications that way?

File: setup/local-preview.md (hunk 1 above)
Comment (khorne3) [reply setup/local-preview.md]: @coadler Do we have a solution for this? I was trying this out while editing and got the same time out error

File: setup/local-preview.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+2. [kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation)
+3. [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl)
+4. [helm](https://helm.sh/docs/intro/install)
+
+## Installing Coder
+
+To install Coder, run:
+
+```bash
+curl -fsSL https://coder.com/try.sh | PORT="8080" sh
```
Comment (Duelsik) [inline setup/local-preview.md:27]: This script is not POSIX-compliant so has to be run using bash, not sh.
(alpine errors out syntax, wsl (debian) errors about pipefail)

Comment (coadler) [reply setup/local-preview.md:27]: Thanks for the report. I'll switch this to bash for now and see if anyone has issues with the requirement of bash over sh.

File: setup/local-preview.md (hunk 1 above)
Comment (coadler) [reply setup/local-preview.md]: @khorne3 sorry, everyone in the office (both mac/linux) is able to run it fine so I haven't been able to figure out what exactly you're running into. While it's waiting for `timescale-0` to come up, could you potentially send me some logs?

* `kubectl describe pod timescale-0` 
* `kubectl logs timecale-0`

should help figure out what's going wrong.

Comment (khorne3) [reply setup/local-preview.md]: Hey @coadler here's the [output](https://docs.google.com/document/d/1ruxNQmnBdvFWi0KsNueOEq9PdLOtXAeWw2BrnI0LHlw/edit?usp=sharing) I'm getting; please let me know if I can provide anything else

Comment (coadler) [reply setup/local-preview.md]: Thanks! I think I see why it's not coming up. I'll get an update pushed out and have you try again in a bit.

File: setup/kubernetes/local-preview.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+If you choose not to try CVMs, these Docker versions are not necessary.
+
+### DevURLs
+
+Currently, the local preview doesn't support [DevURLs][devurl-url]. Tools such
+as [ngrok][ngrok-url] can be used to preview webpages from inside an
+environment.
+
+We are working on bringing DevURL support to local previews in later releases.
```
Comment (coadler) [inline setup/kubernetes/local-preview.md:43]: cc @bpmct

File: setup/local-preview.md (hunk 2 above)
Comment (Duelsik) [reply setup/local-preview.md]: I mean... if script is not POSIX only - normal sh wouldn't be able to run this.

File: setup/local-preview.md (hunk 1 above)
Comment (coadler) [reply setup/local-preview.md]: @khorne3 try giving it another go

File: setup/local-preview.md (hunk 2 above)
Comment (coadler) [reply setup/local-preview.md]: Sorry, I meant going with bash vs removing the pipefail (or other potentially non-POSIX things) to stick with sh.

Comment (coadler) [reply setup/local-preview.md]: This definitely emphasizes getting testing setup to make sure this works correctly in the standard bash shipped with most OSes. I'll definitely try and prioritize that before this gets much marketing push.

File: setup/local-preview.md (hunk 1 above)
Comment (khorne3) [reply setup/local-preview.md]: @coadler Worked this time around. Thanks!

File: setup/local-preview.md (hunk 2 above)
Comment (coadler) [reply setup/local-preview.md]: Hey @Duelsik, pushed an update last night that made the script POSIX compatible and tested it on busybox (alpine) sh. It should now be fully POSIX now.

**Pull request merged**